[Aung Khant Kyaw] iP

AkkFiros opened the pull request.

Commit:
Add Gradle support

AkkFiros committed:
Level - 1

Added in Greet, Echo and Exit.

AkkFiros committed:
Level - 2

Added task addition, task list

AkkFiros committed:
Level - 3

Added in mark as done, mark as undone, icon representation of completion status, custom replies to errors.

AkkFiros committed:
Level - 4

Added in the ToDo, Deadline and Event classes and modified the behavior of KKBot to accept the new keywords.

AkkFiros committed:
Disallowed arbitrary tasks from being accepted

Change the functionality of KKBot to only accept and create tasks that include keywords (todo, deadline, event). Other miscellaneous tasks or inputs without the use of keywords result in an error.

AkkFiros committed:
Updated description

Updated description of error message.

AkkFiros committed:
A-TextUiTesting

Updated files in text-ui-test folder to test KKBot. Cleaned up some code in KKBot.java for visual style.

AkkFiros committed:
Level - 6

Edited and cleaned up code for exception handling.

AkkFiros committed:
Level - 6

Added in delete function for KKBot. Updated A-TextUiTesting files for updated functions and test cases.

jonasgwt commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,68 @@
+     * Obtains and returns the completion status of the task
+     * @return The completion status
+     */
+    public boolean getCompletion() {
```
Perhaps this method name should be phrased to sound more like a boolean for example, `isCompleted`

jonasgwt commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,23 @@
+    }
+
+    /**
+     * toString method for a ToDo task
```
Perhaps we can start our method header comments in the form `Returns ...`, `Sends ...`, `Adds ...`

jonasgwt commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,33 @@
+
+    /**
+     * Constructor for Deadline task
+     * @param description description of a task that the user inputs
```
It would be better to have a full stop at the end of each parameter description.

jonasgwt commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,68 @@
+
+    /**
+     * Obtains and returns the icon representation of the completion status of the task
+     * @return the icon representing the completion status of the task
```
I think it would be better if we start the description of our `@return` with a capital letter.

maxtance commented on src/main/java/KKBot.java:
```diff
@@ -0,0 +1,168 @@
+                            System.out.println(index + ". " + task.toString() + "\n");
+                        }
+                    }
+                }
```
I feel that the code here is deeply nested. You could consider creating different methods to handle different things. For instance, a UI class to handle the printing and a getList method to get the current list of tasks.

maxtance commented on src/main/java/KKBot.java:
```diff
@@ -0,0 +1,168 @@
+                    System.out.println("Woopsies you left out the task number. Please specify one!\n" + divider);
+                } catch (IndexOutOfBoundsException ioobe) {
+                    System.out.println("Hmm... that task doesn't exist. Please input another task number!\n" + divider);
+                }
```
I like the use of try-and-catch blocks. Great job on considering and handling the different exceptions.

maxtance commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,68 @@
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
Good use of access modifiers which follows OOP 👍

maxtance commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,68 @@
+/**
```
You could consider putting Task, ToDo, Deadline, and Event into a package.

maxtance reviewed (commented):
Overall, the basic level for code quality is met. To improve the code quality, I feel that there could be more abstractions, especially in the main method. Keep it up 👍

AkkFiros committed:
Overhaul Level-1 to Level-6

Code is messy and has poor OOP.
It is very hard to make additions/revisions.
Let's separate the code into other classes.
Inheritance allows for better OOP and enables future code edits.

AkkFiros committed:
Level-7

Add Storage class to save KKBot Tasks onto hard drive.

AkkFiros committed:
Level-8

Add date and time functionality to KKBot.

AkkFiros committed:
Level-8

Add date and time functionality to KKBot.

AkkFiros committed:
A-MoreOOP

Extract classes from previous code for better OOP.

AkkFiros committed:
A-Packages

Divide classes into packages.

AkkFiros committed:
Clean up code

Edit some classes for improved readability.

AkkFiros committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

AkkFiros committed:
A-JUnit

AkkFiros committed:
A-Jar

AkkFiros committed:
A-JavaDoc

AkkFiros committed:
A-CodingStandard

AkkFiros committed:
Level-9

AkkFiros committed:
Use GFMD

Update README.md to follow GFMD standards

AkkFiros committed:
Fix bug with date/time

AkkFiros committed:
Fix bugs for JUnit Tests

AkkFiros committed:
Edit UI displays

AkkFiros committed:
A-CheckStyle

AkkFiros committed:
Level-10

Add GUI and chat box window to KKBot.

AkkFiros committed:
Generate new JAR file

AkkFiros committed:
A-Assertions

AkkFiros committed:
A-CodeQuality

AkkFiros committed:
Merge pull request #2 from AkkFiros/branch-A-Assertions

A-Assertions

AkkFiros committed:
Merge branch 'master' into branch-A-CodeQuality

AkkFiros committed:
Merge pull request #3 from AkkFiros/branch-A-CodeQuality

A-CodeQuality

AkkFiros committed:
B-DoAfterTasks

AkkFiros committed:
Merge branch 'branch-B-DoAfterTasks'

AkkFiros committed:
Clean up GUI display

AkkFiros committed:
Update README.md

AkkFiros committed:
Adjust GUI display

AkkFiros committed:
Update representative screenshot
